src/t8_schemes/t8_default/t8_default_common/t8_default_common.hxx

```diff
@@ -24,37 +24,179 @@
  * We provide some functions that are useful across element classes.
  */
 
-#pragma once
+#ifndef T8_DEFAULT_COMMON_HXX
+#define T8_DEFAULT_COMMON_HXX
 
 #include <t8_element.hxx>
+#include <t8_schemes/t8_crtp.hxx>
+#include <sc_functions.h>
 
 /* Macro to check whether a pointer (VAR) to a base class, comes from an
  * implementation of a child class (TYPE). */
 #define T8_COMMON_IS_TYPE(VAR, TYPE) ((dynamic_cast<TYPE> (VAR)) != NULL)
 
-class t8_default_scheme_common_c: public t8_eclass_scheme_c {
+/** This class independent function assumes an sc_mempool_t as context.
+ * It is suitable as the element_new callback in \ref t8_eclass_scheme.
+ * We assume that the mempool has been created with the correct element size.
+ * \param [in,out] ts_context   An element is allocated in this sc_mempool_t.
+ * \param [in]     length       Non-negative number of elements to allocate.
+ * \param [in,out] elem         Array of correct size whose members are filled.
+ */
+inline static void
+t8_default_mempool_alloc (sc_mempool_t *ts_context, int length, t8_element_t **elem)
+{
+  int i;
+
+  T8_ASSERT (ts_context != NULL);
+  T8_ASSERT (0 <= length);
+  T8_ASSERT (elem != NULL);
+
+  for (i = 0; i < length; ++i) {
+    elem[i] = (t8_element_t *) sc_mempool_alloc (ts_context);
+  }
+}
+
+/** This class independent function assumes an sc_mempool_t as context.
+ * It is suitable as the elem_destroy callback in \ref t8_eclass_scheme_t.
+ * We assume that the mempool has been created with the correct element size.
+ * \param [in,out] ts_context   An element is returned to this sc_mempool_t.
+ * \param [in]     length       Non-negative number of elements to destroy.
+ * \param [in,out] elem         Array whose members are returned to the mempool.
+ */
+inline static void
+t8_default_mempool_free (sc_mempool_t *ts_context, int length, t8_element_t **elem)
+{
+  int i;
+
+  T8_ASSERT (ts_context != NULL);
+  T8_ASSERT (0 <= length);
+  T8_ASSERT (elem != NULL);
+
+  for (i = 0; i < length; ++i) {
+    sc_mempool_free (ts_context, elem[i]);
+  }
+}
+
+/* Given an element's level and dimension, return the number of leaves it
+ * produces at a given uniform refinement level */
+static inline t8_gloidx_t
+count_leaves_from_level (int element_level, int refinement_level, int dimension)
+{
+  return element_level > refinement_level ? 0 : sc_intpow64 (2, dimension * (refinement_level - element_level));
```

Comment:
```suggestion
  return element_level > refinement_level ? 0 : 2 << (dimension * (refinement_level - element_level));
```

Reply:
This isnt quite right. But this solution should work